### PR TITLE
SRS-543: saving suspect land uses data

### PR DESCRIPTION
**Pull request by anton-bcgov**
Adds method for preserving updates to Suspect Land Uses in the DB.

### CODE REVIEW

File: backend/sites/src/app/services/landHistory/landHistory.service.ts [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+import { Repository, Brackets, In } from 'typeorm';
 import { LandHistories } from '../../entities/landHistories.entity';
+import { LandHistoriesInputDTO } from 'src/app/dto/landHistoriesInput.dto';
+import { TransactionManagerService } from '../transactionManager/transactionManager.service';
 
 export class LandHistoryService {
   constructor(
     @InjectRepository(LandHistories)
     private landHistoryRepository: Repository<LandHistories>,
+    private transactionManagerService: TransactionManagerService,
```
Comment (jaspalsingh-aot) [inline backend/sites/src/app/services/landHistory/landHistory.service.ts:14]: is this going to create a separate transaction object ? there is one in site service as well

File: frontend/site-registry/src/app/features/details/SiteDetails.tsx [hunk 2]
```diff
@@ -457,8 +457,8 @@ const SiteDetails = () => {
           <ModalDialog
             closeHandler={(response) => {
               setSave(false);
-              if (response) {
-                dispatch(saveSiteDetails(null)).unwrap();
+              if (response && id) {
+                dispatch(saveSiteDetails(id)).unwrap();
```
Comment (jaspalsingh-aot) [inline frontend/site-registry/src/app/features/details/SiteDetails.tsx]: passing id won't don't anything as the parameter is unused

File: backend/sites/src/app/middleware/userJwtTokenDecoder.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ export class UserJWTTokenDecoderMiddleware implements NestMiddleware {
         const { email, sub, identity_provider, given_name, family_name } =
           decodedToken;
 
+        req['user'] = decodedToken;
```
Comment (jaspalsingh-aot) [inline backend/sites/src/app/middleware/userJwtTokenDecoder.ts]: what this code will do here ?

File: backend/sites/src/app/services/landHistory/landHistory.service.ts (hunk 1 above)
Comment (anton-bcgov) [reply backend/sites/src/app/services/landHistory/landHistory.service.ts:14]: No, this service allows sharing the same transaction manager across different part of the app within the same request.

It is set [here](https://github.com/bcgov/nr-site-registry/pull/63/files#diff-d04629285cacd644841e90fe6881c99726c5a4be674b67c8f8f101bbcf93018bR303-R305), the same instance is reused in this file.

File: frontend/site-registry/src/app/features/details/SiteDetails.tsx (hunk 2 above)
Comment (anton-bcgov) [reply frontend/site-registry/src/app/features/details/SiteDetails.tsx]: It is used now:
- [saveSiteDetails thunk](https://github.com/bcgov/nr-site-registry/pull/63/files#diff-f518fb81c96790defb1d1b9776791133e54c53baf0914f7994b22cf460d14101R30)
- [saveSiteDetails of site service](https://github.com/bcgov/nr-site-registry/pull/63/files#diff-d04629285cacd644841e90fe6881c99726c5a4be674b67c8f8f101bbcf93018bR290)

File: backend/sites/src/app/middleware/userJwtTokenDecoder.ts (hunk 3 above)
Comment (anton-bcgov) [reply backend/sites/src/app/middleware/userJwtTokenDecoder.ts]: This sets decoded token info on the request object. It allows accessing [the current user at the time of the request](https://github.com/bcgov/nr-site-registry/pull/63/files#diff-48df5eb045a176ba7be68e1b7c753927d63656242a0d6cf0e3105bd8e46c2ea2R64)

File: backend/sites/src/app/services/landHistory/landHistory.service.ts (hunk 1 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/services/landHistory/landHistory.service.ts:14]: Ok

File: frontend/site-registry/src/app/features/details/SiteDetails.tsx (hunk 2 above)
Comment (jaspalsingh-aot) [reply frontend/site-registry/src/app/features/details/SiteDetails.tsx]: ok

File: backend/sites/src/app/middleware/userJwtTokenDecoder.ts (hunk 3 above)
Comment (jaspalsingh-aot) [reply backend/sites/src/app/middleware/userJwtTokenDecoder.ts]: ok

Comment (midhun-aot) [inline backend/sites/src/app/middleware/userJwtTokenDecoder.ts]: As discussed we already get user information using "@AuthicatedUser()" we don't need to set it again in request. Please remove it.

File: backend/sites/src/app/services/landHistory/landHistory.service.ts [hunk 4]
```diff
@@ -60,4 +67,71 @@ export class LandHistoryService {
...
+            siteId,
+            lutCode: In(deletes),
+          });
+          return null;
+        },
+      );
+
+      return updatedRecords;
+    } catch (e) {
+      console.log('Error updating land histories', e);
```
Comment (midhun-aot) [inline backend/sites/src/app/services/landHistory/landHistory.service.ts:129]: Please throw error as we are in a transaction.

File: frontend/site-registry/src/app/components/table/body/TableBody.tsx [hunk 5]
```diff
@@ -148,6 +148,7 @@ const TableBody: FC<TableBodyProps> = ({
       row: getDataRow(rowKey),
       property: propertyName,
       value: value,
+      index: rowKey,
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/components/table/body/TableBody.tsx]: Please remove it as not required and not used anywhere.

File: frontend/site-registry/src/app/features/details/SaveSiteDetailsSlice.ts [hunk 6]
```diff
@@ -27,8 +27,9 @@ const initialState: SaveSiteDetails = {
 
 export const saveSiteDetails = createAsyncThunk(
   'saveSiteDetails',
-  async (siteDetailsDTO: any, { getState }) => {
+  async (siteId: string, { getState }) => {
     const saveDTO = getSiteDetailsToBeSaved(getState());
+    saveDTO.siteId = siteId;
```
Comment (midhun-aot) [inline frontend/site-registry/src/app/features/details/SaveSiteDetailsSlice.ts]: We already have site we dont't need to pass siteId to it. Please update the code to have no parameters being passed to this function.

**Review by midhun-aot — CHANGES_REQUESTED** [review]

**Review by midhun-aot — APPROVED** [review]

**Pull request merged**